File: LimDaeYoung/be-onboarding-project/common-api/src/main/java/com/survey/api/entity/SurveyOptionEntity.java
package com.survey.api.entity;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Data
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity(name="surveyOption")
public class SurveyOptionEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String optionName;
    private int optionOrder;
    private String useYn;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "itemId")
    private SurveyItemEntity surveyItem; // 설문 항목 참조

    public SurveyOptionEntity(String optionName, int optionOrder, String useYn, SurveyItemEntity surveyItem) {
        this.optionName = optionName;
        this.optionOrder = optionOrder;
        this.useYn = useYn;
        this.surveyItem = surveyItem;
    }
}

File: survey-data/src/main/kotlin/com/ic/surveydata/entity/form/SurveyOptionEntity.kt
package com.ic.surveydata.entity.form

import com.ic.surveydata.entity.BaseTimeEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table

@Entity
@Table(name = "survey_option")
data class SurveyOptionEntity(
    @Id
    @Column(name = "id", nullable = false, unique = true)
    val id: String,
    @Column(nullable = false, unique = false)
    val name: String,
    @ManyToOne
    @JoinColumn(name = "survey_item_id")
    val surveyItemEntity: SurveyItemEntity? = null,
): BaseTimeEntity()
